### PR TITLE
[1LP][RFR] New Test: Testing quota units

### PR DESCRIPTION
**Pull request by ganeshhubale**
## Purpose or Intent
- Testing if assigned qutoa i shown as per units in automation logs

### PRT Run
{{ pytest: cfme/tests/automate/test_common_methods.py::test_automate_quota_units --use-template-cache --use-provider=complete -qsvvv}}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit c9cee64f6e3e803ba56d0bb8b1607bff38928ccb

The *local* fixture **`set_roottenant_quota`** is used in the following files:
- **cfme/tests/automate/test_common_methods.py**
    - *test_automate_quota_units*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/automate/test_common_methods.py [hunk 1]
```diff
@@ -250,3 +253,73 @@ def test_service_retirement_from_automate_method(request, generic_catalog_item,
...
+
+    def _finalize():
+        collection = appliance.provider_based_collection(provider)
+        vm_obj = collection.instantiate(vm_name, provider, provisioning["template"])
+        try:
+            vm_obj.cleanup_on_provider()
+        except Exception:
+            logger.warning('Failed deleting VM from provider: %s', vm_name)
+
+    request.addfinalizer(_finalize)
```
Comment (sbulage) [inline cfme/tests/automate/test_common_methods.py]: **Suggestion:**
Better to use single line `request.addfinalizer()` for inline functions. You can use decorator like this:

```
@requsest.addfinalizer
def _finalize():
    collection = appliance.provider_based_collection(provider)
    vm_obj = collection.instantiate(vm_name, provider, provisioning["template"])
    try:
        vm_obj.cleanup_on_provider()
    except Exception:
        logger.warning('Failed deleting VM from provider: %s', vm_name)
```

File: cfme/tests/automate/test_common_methods.py [hunk 2]
```diff
@@ -250,3 +253,73 @@ def test_service_retirement_from_automate_method(request, generic_catalog_item,
...
+            "/var/www/miq/vmdb/log/automation.log",
+            matched_patterns=[f'.*Getting Tenant Quota Values for:.*.memory=>1073741824000.*'],
+    ).waiting(timeout=120):
+        # Provisioning VM via lifecycle
+        do_vm_provisioning(appliance, template_name=provisioning["template"], provider=provider,
+                           vm_name=vm_name, provisioning_data=prov_data, wait=False, request=None)
+
+        # nav to requests page to check quota validation
+        request_description = 'Provision from [{template}] to [{vm}{msg}]'.format(
+            template=provisioning["template"], vm=vm_name, msg='')
```
Comment (sbulage) [inline cfme/tests/automate/test_common_methods.py]: **Question:**
Why `msg=''` is empty.?

File: cfme/tests/automate/test_common_methods.py [hunk 3]
```diff
@@ -250,3 +253,73 @@ def test_service_retirement_from_automate_method(request, generic_catalog_item,
         description=f"Service Retire for: {generic_catalog_item.name}")
     wait_for(lambda: retire_request.request_state == "finished", fail_func=retire_request.reload,
              timeout=180, delay=10)
+
+
+@pytest.fixture
+def set_roottenant_quota(request, appliance):
```
Comment (sbulage) [inline cfme/tests/automate/test_common_methods.py]: **Suggestion:**
Can you rename it to something like: `set_root_tenant_quota` .?

File: cfme/tests/automate/test_common_methods.py [hunk 4]
```diff
@@ -250,3 +253,71 @@ def test_service_retirement_from_automate_method(request, generic_catalog_item,
...
+    with LogValidator(
+            "/var/www/miq/vmdb/log/automation.log",
+            matched_patterns=[f'.*Getting Tenant Quota Values for:.*.memory=>1073741824000.*'],
+    ).waiting(timeout=120):
+        # Provisioning VM via lifecycle
+        do_vm_provisioning(appliance, template_name=provisioning["template"], provider=provider,
+                           vm_name=vm_name, provisioning_data=prov_data, wait=False, request=None)
+
+        # nav to requests page to check quota validation
+        request_description = f'Provision from [{provisioning["template"]}] to [{vm_name}]'
```
Comment (valaparthvi) [inline cfme/tests/automate/test_common_methods.py]: I've sometimes seen that request description changes for different appliance versions. If that happens with you, you might want to find a solution to that.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit bf6b5eb37c2276c4e7d43733253d31fffdfa1368

The *local* fixture **`set_root_tenant_quota`** is used in the following files:
- **cfme/tests/automate/test_common_methods.py**
    - *test_automate_quota_units*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/automate/test_common_methods.py [hunk 5]
```diff
@@ -250,3 +253,71 @@ def test_service_retirement_from_automate_method(request, generic_catalog_item,
...
              timeout=180, delay=10)
+
+
+@pytest.fixture
+def set_root_tenant_quota(request, appliance):
+    field, value = request.param
+    root_tenant = appliance.collections.tenants.get_root_tenant()
+    root_tenant.set_quota(**{'{}_cb'.format(field): True, field: value})
+    yield
+    root_tenant.set_quota(**{'{}_cb'.format(field): False})
```
[the remark this answers is not in the document]
Comment (ganeshhubale) [reply cfme/tests/automate/test_common_methods.py]: I am not disabling all the values. Just disabled the value which I had assigned. I feel this is proper cleanup for that particular quota set.

File: cfme/tests/automate/test_common_methods.py [hunk 6]
```diff
@@ -250,3 +253,73 @@ def test_service_retirement_from_automate_method(request, generic_catalog_item,
...
+def set_roottenant_quota(request, appliance):
+    field, value = request.param
+    roottenant = appliance.collections.tenants.get_root_tenant()
+    roottenant.set_quota(**{'{}_cb'.format(field): True, field: value})
+    yield
+    roottenant.set_quota(**{'{}_cb'.format(field): False})
+
+
+@pytest.mark.tier(3)
+@pytest.mark.usefixtures("setup_provider")
```
[the remark this answers is not in the document]
Comment (ganeshhubale) [reply cfme/tests/automate/test_common_methods.py]: No proper reason. We have these two ways for using this fixture. 
But I would like to know how will it affect test case?

File: cfme/tests/automate/test_common_methods.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (valaparthvi) [reply cfme/tests/automate/test_common_methods.py]: What I meant was, you might want to reset the values to how they were before.

File: cfme/tests/automate/test_common_methods.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (valaparthvi) [reply cfme/tests/automate/test_common_methods.py]: I don't think they affect the test in any way.

File: cfme/tests/automate/test_common_methods.py (hunk 4 above)
Comment (ganeshhubale) [reply cfme/tests/automate/test_common_methods.py]: Actually, I have not seen this type of issue with request description of VM provisioning. 
Please let me know the issue related to it and solution you have performed.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit dac568e6da64de7a7e4f5c7bb7cd34b3211248b9

The *local* fixture **`set_root_tenant_quota`** is used in the following files:
- **cfme/tests/automate/test_common_methods.py**
    - *test_automate_quota_units*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/automate/test_common_methods.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ganeshhubale) [reply cfme/tests/automate/test_common_methods.py]: I am only changing one value and again cleaning up that value. So there is no harm to other quota set.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 09cc65cfe4c4504b34d8c2c4efcde12d43029cef

The *local* fixture **`set_root_tenant_quota`** is used in the following files:
- **cfme/tests/automate/test_common_methods.py**
    - *test_automate_quota_units*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/automate/test_common_methods.py (hunk 4 above)
Comment (valaparthvi) [reply cfme/tests/automate/test_common_methods.py]: I faced similar issue while dealing with VM retirement request.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 9bddb733ace02b1eead860f6fe41e2bb8159a50e

The *local* fixture **`set_root_tenant_quota`** is used in the following files:
- **cfme/tests/automate/test_common_methods.py**
    - *test_automate_quota_units*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Pull request merged**